pkg/virt-controller/services/template.go

```diff
@@ -1319,29 +1296,22 @@ func (t *templateService) renderLaunchManifest(vmi *v1.VirtualMachineInstance, i
 			resources.Limits[k8sv1.ResourceCPU] = resource.MustParse("200m")
 			resources.Limits[k8sv1.ResourceMemory] = resource.MustParse("64M")
 		}
-		sidecar := k8sv1.Container{
-			Name:            fmt.Sprintf("hook-sidecar-%d", i),
-			Image:           requestedHookSidecar.Image,
-			ImagePullPolicy: requestedHookSidecar.ImagePullPolicy,
-			Command:         requestedHookSidecar.Command,
-			Args:            requestedHookSidecar.Args,
-			Resources:       resources,
-			SecurityContext: &k8sv1.SecurityContext{
-				RunAsUser:  &userId,
-				Privileged: &privileged,
-			},
-			VolumeMounts: []k8sv1.VolumeMount{
-				{
-					Name:      hookSidecarSocks,
-					MountPath: hooks.HookSocketsSharedDirectory,
-				},
-			},
+
+		sidecarOpts := []Option{
+			WithResourceRequirements(resources),
+			WithVolumeMounts(sidecarVolumeMount()),
+			WithArgs(requestedHookSidecar.Args),
 		}
-		if nonRoot {
-			sidecar.SecurityContext.RunAsGroup = &userId
-			sidecar.SecurityContext.RunAsNonRoot = &nonRoot
+
+		if userId != 0 {
```

Comment:
can you keep using `nonRoot` for consistency? (there's another condition that uses it at line 1137)

Reply:
Good catch. I forgot this variable was still available :) I just need to use this condition on the "renderer". 

Thanks for spotting it.

Reply:
np, you have my virtual lgtm so on, do you plan to continue working on this PR to address the `Other things that might be relevant ` or will you do those things in a separate one? I'm asking because you marked this as a draft.

Reply:
... still trying to figure that out. 

Each of those topics is a world on its own, so I'll probably split these into different PRs. Also, that way we can divide and conquer. The more, the merrier. 

I'll provide a more definitive reply by Monday - but my current inclination is to shoot this PR with this not-so-ambitious scope.

Reply:
I'll at least check out if I can get the environment variables in.

Reply:
> ... still trying to figure that out.
> 
> Each of those topics is a world on its own, so I'll probably split these into different PRs. Also, that way we can divide and conquer. The more, the merrier.
I'd prefer multiple PRs as well, they're probably faster to get in too.

> 
> I'll provide a more definitive reply by Monday - but my current inclination is to shoot this PR with this not-so-ambitious scope.